src/qibolab/pulses/pulse.py

```diff
@@ -27,7 +27,13 @@ class PulseType(Enum):
     VIRTUALZ = "vz"
 
 
-class Pulse(Model):
+class _PulseLike(Model):
```

Comment:
Why don't we make this "public"
```suggestion
class PulseLike(Model):
```
to replace the type defined some lines below in https://github.com/qiboteam/qibolab/blob/1e214df131e69ab97f01c110b6c07da3e2b7086f/src/qibolab/pulses/pulse.py#L131

Reply:
I tried. It seemed a clever idea, and I did it immediately. But Pydantic doesn't like it during deserialization.

It makes sense: a base class does not know which are its child classes. So, if you only specify that, you don't know in practice what to instantiate.
The parent class is a lower bound ("my class is a child of this"), while to deserialize you need at least an upper bound ("my class is one of these").

However, we can decide about possible renaming, and what to make public. For the time being, I decided not to introduce any other feature: the union was already there and exposed, and this has the only net effect of extending the `.id` support to the delay and virtual Z.